next_generation/pint_server/app.py

```diff
@@ -57,33 +58,20 @@ def get_supported_providers():
     global CACHE_PROVIDERS
 
     if CACHE_PROVIDERS is None:
-        CACHE_PROVIDERS = query_providers()
+        versions  = VersionsModel.query.with_entities(VersionsModel.tablename)
+        providers = []
+        for version in versions:
+            provider = re.sub('(servers|images)', '', version.tablename)
+            if provider not in providers:
+                providers.append(provider)
+        CACHE_PROVIDERS = providers
```

Comment:
Well caching is somewhat of an odd concept to use in the way we are deploying the code. Every query will trigger the lambda function and each time it runs it starts from scratch, i.e. it is a stateless implementation. As such a cache is only usefull if in the same execution of a query the providers list needs to be accessed multiple times.

Reply:
When I first implemented this, I was thinking the possibility of running this code outside of AWS Lambda. James mentioned that we want this to be portable not just between cloud providers, but we want it to be able to independently hosted by an open source project. This Flask app can also be running with Apache2 WSGI.